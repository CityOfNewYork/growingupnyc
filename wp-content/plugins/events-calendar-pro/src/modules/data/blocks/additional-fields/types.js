/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const SET_ADDITIONAL_FIELD_INITIAL_STATE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_INITIAL_STATE`;
export const ADD_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/ADD_ADDITIONAL_FIELD`;
export const REMOVE_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD`;

export const SET_ADDITIONAL_FIELD_LABEL = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_LABEL`;
export const SET_ADDITIONAL_FIELD_IS_PRISTINE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_IS_PRISTINE`;
export const SET_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_TYPE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_TYPE`;
export const SET_ADDITIONAL_FIELD_OPTIONS = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OPTIONS`;
export const SET_ADDITIONAL_FIELD_DIVIDER_LIST = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_LIST`;
export const SET_ADDITIONAL_FIELD_DIVIDER_END = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_END`;
export const APPEND_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/APPEND_ADDITIONAL_FIELD_VALUE`;
export const REMOVE_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_CHANGE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_CHANGE`;
export const SET_ADDITIONAL_FIELD_META_KEY = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_META_KEY`;
export const SET_ADDITIONAL_FIELD_OUTPUT = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OUTPUT`;
export const SET_ADDITIONAL_FIELD_BLUR = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_BLUR`;
