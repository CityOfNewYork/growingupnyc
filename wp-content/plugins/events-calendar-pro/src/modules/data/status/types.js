/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const SET_SERIES_QUEUE_STATUS = `${ PREFIX_EVENTS_PRO_STORE }/SET_SERIES_QUEUE_STATUS`;
