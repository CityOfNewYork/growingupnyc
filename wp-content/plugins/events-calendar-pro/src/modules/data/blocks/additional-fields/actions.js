/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { SET_ADDITIONAL_FIELD_OUTPUT } from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_ADDITIONAL_FIELD_INITIAL_STATE,
	payload: props,
} );

export const setFieldChange = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_CHANGE,
	payload: {
		name,
	},
} );

export const setFieldBlur = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_BLUR,
	payload: {
		name,
	},
} );

export const setFieldBlurWithType = ( name, type ) => ( {
	type: `${ types.SET_ADDITIONAL_FIELD_BLUR }/${ type }`,
	payload: {
		name,
	},
} );

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const appendFieldValue = ( name, value ) => ( {
	type: types.APPEND_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const removeFieldValue = ( name, value ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

export const setFieldMetaKey = ( name, metaKey ) => ( {
	type: types.SET_ADDITIONAL_FIELD_META_KEY,
	payload: {
		name,
		metaKey,
	},
} );

export const setFieldOutput = ( name, output ) => ( {
	type: SET_ADDITIONAL_FIELD_OUTPUT,
	payload: {
		name,
		output,
	},
} );
