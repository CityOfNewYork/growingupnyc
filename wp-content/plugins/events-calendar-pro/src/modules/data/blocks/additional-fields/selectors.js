/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
