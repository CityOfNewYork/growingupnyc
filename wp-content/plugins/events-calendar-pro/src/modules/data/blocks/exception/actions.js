/**
 * External dependencies
 */
import { curry } from 'lodash/fp';

/**
 * Internal dependencies
 */
import * as types from './types';

export const addField = () => ( {
	type: types.ADD_EXCEPTION_FIELD,
} );

export const addException = ( payload ) => ( {
	type: types.ADD_EXCEPTION,
	payload,
} );

export const removeField = () => ( {
	type: types.REMOVE_EXCEPTION_FIELD,
} );

export const removeException = ( index ) => ( {
	type: types.REMOVE_EXCEPTION,
	index,
} );

export const editException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
} ) );

export const syncException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
	sync: true,
} ) );

export const syncExceptionsFromDB = payload => ( {
	type: types.SYNC_EXCEPTIONS_FROM_DB,
	payload,
} );
