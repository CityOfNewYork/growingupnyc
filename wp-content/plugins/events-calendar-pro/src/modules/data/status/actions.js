/**
 * Internal dependencies
 */
import * as types from './types';

export const setSeriesQueueStatus = ( payload ) => ( {
	type: types.SET_SERIES_QUEUE_STATUS,
	payload,
} );
