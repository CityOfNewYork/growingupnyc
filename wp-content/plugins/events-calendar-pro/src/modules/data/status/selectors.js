/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getStatus = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].status;

export const isCompleted = createSelector( getStatus, status => !! status.done );
export const getProgress = createSelector( getStatus, status => status.progress );
