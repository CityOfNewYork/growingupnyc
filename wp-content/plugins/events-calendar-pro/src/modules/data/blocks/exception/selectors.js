/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	EXCEPTION_OCCURRENCE_OPTIONS,
} from './options';
import { SERIES_ENDS_OPTIONS } from '@moderntribe/events-pro/data/blocks/recurring/options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getExceptions = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.exception;
export const getRules = getExceptions;
export const getExceptionsCount = createSelector( getExceptions, ( exceptions ) => exceptions.length );
export const hasExceptions = createSelector( getExceptionsCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getExceptions, getIndex ],
	( exceptions, index ) => exceptions[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( EXCEPTION_OCCURRENCE_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
