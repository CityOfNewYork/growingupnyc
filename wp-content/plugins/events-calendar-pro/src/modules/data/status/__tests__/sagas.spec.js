/**
 * External dependencies
 */
import { sprintf } from '@wordpress/i18n';
import { some, noop } from 'lodash';
import { race, take, select, call, put, fork } from 'redux-saga/effects';
import { delay, eventChannel } from 'redux-saga';
import { select as wpSelect, subscribe, dispatch as wpDispatch } from '@wordpress/data';
import 'whatwg-fetch';
import { blocks } from '@moderntribe/events/data';
import { moment as momentUtils } from '@moderntribe/common/utils';

/**
 * Internal dependencies
 */
import * as recurringTypes from '@moderntribe/events-pro/data/blocks/recurring/types';
import { hasRules } from '@moderntribe/events-pro/data/blocks/recurring/selectors';
import * as actions from '../actions';
import * as selectors from '../selectors';
import watchers, * as sagas from '../sagas';

function mock() {
	return {
		select: jest.fn( () => ( {
			isSavingPost: noop,
			isPublishingPost: noop,
			getCurrentPostId: jest.fn( () => 1 ),
		} ) ),
		subscribe: jest.fn( () => noop ),
		dispatch: jest.fn( () => ( {
			removeNotice: noop,
			createSuccessNotice: noop,
		} ) ),
	};
}

jest.mock( '@wordpress/data', () => mock() );

describe( 'Status sagas', () => {
	describe( 'fetchStatus', () => {
		let _FormData, _Response, append, json;

		function FormDataMock() {
			this.append = append;
		}

		function ResponseMock() {
			this.json = json;
		}

		beforeAll( () => {
			_FormData = global.FormData;
			global.FormData = FormDataMock;
			global.Response = ResponseMock;
			global.tribe_js_config = {
				rest: {
					nonce: {
						queue_status_nonce: 'nonce',
					},
				},
			};
			global.ajaxurl = 'https://cool.com';
		} );

		beforeEach( () => {
			append = jest.fn();
			json = jest.fn();
		} );

		afterAll( () => {
			global.FormData = _FormData;
			global.Response = _Response;
			delete global.tribe_js_config;
			delete global.ajaxurl;
		} );

		it( 'should fetch successfully', () => {
			const gen = sagas.fetchStatus();

			expect( gen.next().value ).toMatchSnapshot();
			expect( append.mock.calls ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( append.mock.calls ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( append.mock.calls ).toMatchSnapshot();

			expect( gen.next().value ).toEqual(
				call( global.fetch, global.ajaxurl, {
					method: 'POST',
					credentials: 'same-origin',
					body: new FormData(),
				} )
			);

			const response = new Response();
			expect( gen.next( response ).value ).toEqual(
				call( [ response, 'json' ] )
			);
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should handle errors', () => {
			const gen = sagas.fetchStatus();
			gen.next().value;
			expect( gen.throw().value ).toEqual( false );
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'pollUntilSeriesCompleted', () => {
		it( 'should poll if not done', () => {
			const gen = sagas.pollUntilSeriesCompleted();

			expect( gen.next().value ).toEqual(
				put( blocks.datetime.actions.disableEdits() )
			);

			expect( gen.next().value ).toEqual(
				call( sagas.fetchStatus )
			);

			const response = { done: false, items_created: 1, last_created_at: '2019-11-07 00:00:00' };
			expect( gen.next( response ).value ).toEqual(
				put( actions.setSeriesQueueStatus( response ) )
			);

			expect( gen.next().value ).toEqual(
				 call(
					[ wpDispatch( 'core/notices' ), 'createSuccessNotice' ],
					`${ sprintf( sagas.NOTICES[ sagas.NOTICE_PROGRESS_ON_SERIES_CREATION_COUNT ], response.items_created ) } ${ sprintf( sagas.NOTICES[ sagas.NOTICE_PROGRESS_ON_SERIES_CREATION ], momentUtils.toDate( momentUtils.toMoment( response.last_created_at ) ) ) }`,
					{ id: sagas.NOTICE_PROGRESS_ON_SERIES_CREATION, isDismissible: true }
				)
			);

			expect( gen.next( false ).value ).toEqual(
				select( selectors.isCompleted )
			);

			expect( gen.next().value ).toEqual(
				call( delay, 1000 )
			);

			expect( gen.next().done ).toBeFalsy();
		} );
		it( 'should exit when done', () => {
			const gen = sagas.pollUntilSeriesCompleted();

			expect( gen.next().value ).toEqual(
				put( blocks.datetime.actions.disableEdits() )
			);

			expect( gen.next().value ).toEqual(
				call( sagas.fetchStatus )
			);

			const response = { done: true, items_created: 3, last_created_at: '2019-11-07 00:00:00' };
			expect( gen.next( response ).value ).toEqual(
				put( actions.setSeriesQueueStatus( response ) )
			);

			expect( gen.next( true ).value ).toEqual(
				select( selectors.isCompleted )
			);

			expect( gen.next( true ).value ).toEqual(
				put( blocks.datetime.actions.allowEdits() )
			);

			expect( gen.next().value ).toEqual(
				call(
					[ wpDispatch( 'core/editor' ), 'removeNotice' ],
					sagas.NOTICE_EDITING_SERIES
				)
			);

			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'createWPEditorChannel', () => {
		it( 'should create channel', () => {
			expect( sagas.createWPEditorChannel() ).toMatchSnapshot();
		} );
	} );

	describe( 'actionTaker', () => {
		it( 'just takes actions', () => {
			const gen = sagas.actionTaker();

			expect( gen.next().value ).toEqual(
				take( [ recurringTypes.SYNC_RULES_FROM_DB ] )
			);

			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'showEditingAllSeriesPrompt', () => {
		it( 'should show prompt', () => {
			const gen = sagas.showEditingAllSeriesPrompt();

			expect( gen.next().value ).toEqual(
				take( [ recurringTypes.SYNC_RULES_FROM_DB ] )
			);

			expect( gen.next().value ).toEqual(
				select( hasRules )
			);

			expect( gen.next( true ).value ).toEqual(
				call( [ /action=edit/, 'test' ], window.location.search )
			);

			expect( gen.next( true ).value ).toEqual(
				call(
					[ wpDispatch( 'core/notices' ), 'createSuccessNotice' ],
					sagas.NOTICES[ sagas.NOTICE_EDITING_SERIES ],
					{ id: sagas.NOTICE_EDITING_SERIES, isDismissible: false }
				)
			);
		} );
	} );

	describe( 'watchers', () => {
		it( 'should watch for channel or action changes', () => {
			const gen = watchers();

			expect( gen.next().value ).toEqual(
				fork( sagas.showEditingAllSeriesPrompt )
			);

			expect( gen.next().value ).toEqual(
				call( sagas.createWPEditorChannel )
			);

			expect( gen.next( 'Channel' ).value ).toEqual(
				race( [
					take( 'Channel' ),
					call( sagas.actionTaker ),
				] )
			);

			expect( gen.next( 'Channel' ).value ).toEqual(
				call( sagas.pollUntilSeriesCompleted )
			);

			expect( gen.next().done ).toBeFalsy();
		} );
	} );
} );
