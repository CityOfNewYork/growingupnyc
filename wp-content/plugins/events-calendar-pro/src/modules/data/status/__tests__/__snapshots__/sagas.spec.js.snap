// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status sagas createWPEditorChannel should create channel 1`] = `
Object {
  "close": [Function],
  "flush": [Function],
  "take": [Function],
}
`;

exports[`Status sagas fetchStatus should fetch successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "action",
      "gutenberg_events_pro_recurrence_queue",
    ],
    "context": FormDataMock {
      "append": [MockFunction],
    },
    "fn": [MockFunction],
  },
}
`;

exports[`Status sagas fetchStatus should fetch successfully 2`] = `Array []`;

exports[`Status sagas fetchStatus should fetch successfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "recurrence_queue_status_nonce",
      "nonce",
    ],
    "context": FormDataMock {
      "append": [MockFunction],
    },
    "fn": [MockFunction],
  },
}
`;

exports[`Status sagas fetchStatus should fetch successfully 4`] = `Array []`;

exports[`Status sagas fetchStatus should fetch successfully 5`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "post_id",
      1,
    ],
    "context": FormDataMock {
      "append": [MockFunction],
    },
    "fn": [MockFunction],
  },
}
`;

exports[`Status sagas fetchStatus should fetch successfully 6`] = `Array []`;
